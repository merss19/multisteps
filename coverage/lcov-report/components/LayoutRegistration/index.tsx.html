<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/LayoutRegistration/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/LayoutRegistration</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>32/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">56.25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>8/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">94.12% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>32/34</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react'
import {connect} from 'react-redux'
import Background from '../Background'
import Lo<span class="branch-1 cbranch-no" title="branch not covered" >goLink from '../LogoLink'<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="branch-2 cbranch-no" title="branch not covered" ></span></span></span></span>
import *<span class="fstat-no" title="function not covered" ><span class="branch-3 cbranch-no" title="branch not covered" > a</span>s SVGInjector fr<span class="cstat-no" title="statement not covered" >om 'svg-injector'<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span>
console.log(process)
console.log('processs')
interface Sprite {
	readonly body: any<span class="branch-0 cbranch-no" title="branch not covered" ></span>
	readonly img: HTMLImageElement
}
let url = process.env.NODE_ENV === 'test' ? 'https://lk.todayme.ru/' : '/'
class LayoutRegistration extends React.Component&lt;{}, {}&gt; {
&nbsp;
	componentDidMount() {
		this.svgInject()
	}
&nbsp;
	svgInject(): void {<span class="branch-1 cbranch-no" title="branch not covered" ></span>
		const sprite: Sprite = {
			body: document.querySelector('body'),
			img : document.createElement('img')
		}
		sprite.img.className = 'injected-svg'
		sprite.img.id = 'svg-inject-me'
		sprite.img.src = url + 'assets/img/symbol-sprite.svg'
		sprite.body.appendChild(sprite.img)
		let mySVGsToInject: any = document.querySelector('#svg-inject-me')
		SVGInjector(mySVGsToInject)
	}
&nbsp;
	render() {
		const {children} = this.props
		return (
			&lt;div className='layout layout--entry'&gt;
				&lt;div className='grid entry-header'&gt;
					&lt;div className='grid__cell todayme-logo'&gt;
						&lt;LogoLink/&gt;
					&lt;/div&gt;
				&lt;/div&gt;
				{children}
				&lt;Background /&gt;
			&lt;/div&gt;
		)
	}
}
&nbsp;
export default LayoutRegistration</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Sep 20 2017 15:40:06 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
