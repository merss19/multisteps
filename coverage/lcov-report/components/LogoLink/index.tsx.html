<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/LogoLink/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">components/LogoLink</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">38.1% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/21</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">38.1% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/21</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react'
import {browserHistory} from 'react-router'
import cookie from 'react-cookie'
&nbsp;
interface OwnProps {
	isTest?: boolean,
}
<span class="fstat-no" title="function not covered" ></span>
const LogoLi<span class="cstat-no" title="statement not covered" >nk: React.SFC&lt;OwnProps&gt; = ({isTest}) =&gt; (</span>
	&lt;svg classN<span class="cstat-no" title="statement not covered" >ame='svg-icon ys-logo-web'</span>
		 onClick={<span class="cstat-no" title="statement not covered" >() =&gt; {</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('token', {path: '/'})</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('txId', {path: '/'})</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('role', {path: '/'})</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('program', {path: '/'})</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('packageType', {path: '/'})</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('promoName', {path: '/'})</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('share', {path: '/'})</span>
			 cookie.r<span class="cstat-no" title="statement not covered" >emove('userProgram', {path: '/'})</span>
			 cookie.remov<span class="cstat-no" title="statement not covered" >e('tester', {path: '/'})</span>
			 cookie.remove('abtest', {path: '/'})
			 if (isTest) {
				 window.loca<span class="cstat-no" title="statement not covered" >tion.href = 'https://gohero.todayme.ru/'</span>
			 } else {
				 browserHistory.push('/')
			 }
&nbsp;
		 }}&gt;
		&lt;use xlinkHref='#ys-logo-web'/&gt;
	&lt;/svg&gt;)
&nbsp;
export default LogoLink
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Sep 20 2017 15:40:06 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
