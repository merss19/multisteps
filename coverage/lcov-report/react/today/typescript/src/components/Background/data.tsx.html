<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react/today/typescript/src/components/Background/data.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\..\index.html">All files</a> / <a href="index.html">react/today/typescript/src/components/Background</a> data.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react'
&nbsp;
export const bg = &lt;svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1440 1024" className='entry__svg' preserveAspectRatio="xMidYMid slice"&gt;
	&lt;defs&gt;
		&lt;linearGradient id="a" x1="2.235%" y1=".835%" y2="100%"&gt;
			&lt;stop className="entry__bg-stop1" offset="0%"/&gt;
			&lt;stop className="entry__bg-stop2" offset="100%"/&gt;
		&lt;/linearGradient&gt;
	&lt;/defs&gt;
	&lt;g fill="none" fillRule="evenodd"&gt;
		&lt;rect width="1440" height="1024" fill="url(#a)"/&gt;
		&lt;g opacity=".878" transform="translate(144 139)"&gt;
			&lt;path fill="#7FC6F6"
						d="M665.57495,517.992055 C665.57495,508.985055 672.874172,501.683438 681.882642,501.683438 L681.882642,501.683438 C690.889132,501.683438 698.190334,508.986707 698.190334,517.992055 L698.190334,572.351565 C698.190334,581.358566 690.891112,588.660182 681.882642,588.660182 L681.882642,588.660182 C672.876152,588.660182 665.57495,581.356914 665.57495,572.351565 L665.57495,517.992055 Z M936.321197,545.501284 C979.561335,549.307177 1013.47239,585.614574 1013.47239,629.846599 L1013.47239,884.508649 C1013.47239,931.273244 975.565501,969.183438 928.799919,969.183438 L674.144852,969.183438 C627.38154,969.183438 589.472386,931.275514 589.472386,884.508649 L589.472386,629.846599 C589.472386,585.616488 623.381761,549.307029 666.623925,545.501243 C668.953379,551.662325 674.906455,556.043903 681.882642,556.043903 C688.984228,556.043903 695.023596,551.505922 697.262183,545.17181 L905.682919,545.17181 C907.922053,551.505548 913.962232,556.043903 921.062129,556.043903 C928.039967,556.043903 933.992283,551.662697 936.321197,545.501284 Z M904.754437,517.992055 C904.754437,508.985055 912.053659,501.683438 921.062129,501.683438 L921.062129,501.683438 C930.068619,501.683438 937.369821,508.986707 937.369821,517.992055 L937.369821,572.351565 C937.369821,581.358566 930.070599,588.660182 921.062129,588.660182 L921.062129,588.660182 C912.055639,588.660182 904.754437,581.356914 904.754437,572.351565 L904.754437,517.992055 Z M709.062129,670.20088 C709.062129,661.194143 716.364999,653.89274 725.371614,653.89274 L763.421104,653.89274 L763.421104,686.50902 L709.062129,686.50902 L709.062129,670.20088 Z M774.292898,653.89274 L828.651873,653.89274 L828.651873,686.50902 L774.292898,686.50902 L774.292898,653.89274 Z M839.523668,653.89274 L877.573157,653.89274 C886.580637,653.89274 893.882642,661.192163 893.882642,670.20088 L893.882642,686.50902 L839.523668,686.50902 L839.523668,653.89274 Z M709.062129,697.381113 L763.421104,697.381113 L763.421104,729.997392 L709.062129,729.997392 L709.062129,697.381113 Z M839.523668,697.381113 L893.882642,697.381113 L893.882642,729.997392 L839.523668,729.997392 L839.523668,697.381113 Z M709.062129,740.869485 L763.421104,740.869485 L763.421104,773.485764 L709.062129,773.485764 L709.062129,740.869485 Z M839.523668,740.869485 L893.882642,740.869485 L893.882642,773.485764 L839.523668,773.485764 L839.523668,740.869485 Z M839.523668,784.357857 L893.882642,784.357857 L893.882642,816.974136 L839.523668,816.974136 L839.523668,784.357857 Z M709.062129,827.846229 L763.421104,827.846229 L763.421104,860.462508 L725.371614,860.462508 C716.364134,860.462508 709.062129,853.163085 709.062129,844.154368 L709.062129,827.846229 Z M741.677514,784.357857 L796.036488,784.357857 L796.036488,816.974136 L741.677514,816.974136 L741.677514,784.357857 Z M839.523668,827.846229 L893.882642,827.846229 L893.882642,844.154368 C893.882642,853.161105 886.579772,860.462508 877.573157,860.462508 L839.523668,860.462508 L839.523668,827.846229 Z M774.292898,740.869485 L828.651873,740.869485 L828.651873,773.485764 L774.292898,773.485764 L774.292898,740.869485 Z"
						opacity=".723" transform="rotate(30 801.472 735.433)"/&gt;
			&lt;path stroke="#7FC6F6"
						d="M27.36 61.1884615C-.48 78.9565934-8.16 115.933516 9.6 144.266484 27.36 172.119231 63.36 180.282967 90.72 162.034615 97.44 157.712637 103.2 151.95 107.52 145.226923L155.04 221.101648 184.756206 201.827525 108.48 78.9565934C90.72 51.1038462 54.72 42.9401099 27.36 61.1884615zM71.52 15.0873626C73.92 12.6862637 76.32 10.7653846 79.2 8.84450549 103.68-7.48296703 135.84-.27967033 151.68 24.6917582L220.8 134.181868 185.28 201.412637"
						strokeLinecap="round" strokeLinejoin="round"/&gt;
			&lt;path stroke="#7FC6F6" d="M21.6,64.55 L71.52,15.0873626"/&gt;
			&lt;path stroke="#7FC6F6" d="M162.24 41.0192308L118.56 95.2840659M177.12 64.55L136.8 124.097253"
						strokeLinecap="round" strokeLinejoin="round"/&gt;
			&lt;polyline stroke="#7FC6F6"
								points="0 81.844 42.757 81.844 64.974 0 89.788 138.063 117.249 50.923 139.301 111.891 164.283 7.823 185.28 125.702 214.809 41.462 228.737 80.801 268.8 80.118"
								transform="translate(129 454)" strokeLinecap="round" strokeLinejoin="round"/&gt;
		&lt;/g&gt;
		&lt;path stroke="#7FC6F6"
					d="M152.16,53.18 C142.01,44.44 127.82,40 110.01,40 C104.16,40 97.81,41.79 92.72,43.23 C91.81,43.48 90.9,43.74 90.02,43.98 C90.56,25.14 106.06,10 125,10 C127.76,10 130,7.76 130,5 C130,2.24 127.76,-1.13686838e-13 125,-1.13686838e-13 C100.53,-1.13686838e-13 80.55,19.64 80.01,43.99 C79.13,43.75 78.21,43.48 77.29,43.23 C72.19,41.78 65.85,40 60,40 C42.18,40 28,44.43 17.85,53.18 C6,63.38 6.82121026e-13,79.13 6.82121026e-13,100 C6.82121026e-13,121.43 6.24,145.88 16.7,165.39 C28.66,187.71 44.04,200 60,200 C65.17,200 69.33,198.66 73.36,197.36 C77.11,196.15 80.66,195 85,195 C89.34,195 92.89,196.14 96.64,197.36 C100.66,198.66 104.83,200 110,200 C125.96,200 141.34,187.71 153.3,165.39 C163.76,145.87 170,121.43 170,100 C170.01,79.13 164.01,63.38 152.16,53.18 L152.16,53.18 Z"
					transform="rotate(-16 1698.151 -3981.984)"/&gt;
	&lt;/g&gt;
&lt;/svg&gt;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 27 2017 12:25:13 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\..\sorter.js"></script>
</body>
</html>
