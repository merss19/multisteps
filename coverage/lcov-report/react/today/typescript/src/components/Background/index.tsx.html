<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react/today/typescript/src/components/Background/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\..\..\index.html">All files</a> / <a href="index.html">react/today/typescript/src/components/Background</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>16/20</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.14% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>8/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>16/18</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span></td><td class="text"><pre class="prettyprint lang-js">import * as React from 'react'
import {bg} from './data'
&nbsp;
class Bac<span class="branch-1 cbranch-no" title="branch not covered" >kground extends React.Component&lt;{},</span> {}<span class="fstat-no" title="function not covered" ><span class="branch-2 cbranch-no" title="branch not covered" >&gt; </span>{<span class="cstat-no" title="statement not covered" ></span></span>
	render(<span class="fstat-no" title="function not covered" ><span class="branch-3 cbranch-no" title="branch not covered" >) </span>{<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ></span></span></span></span>
		return (
			&lt;div className='entry__bg'&gt;
				{bg}
			&lt;/div&gt;<span class="branch-0 cbranch-no" title="branch not covered" ></span>
		)
	}
}
&nbsp;
export default Background</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 27 2017 12:25:13 GMT+0300 (RTZ 2 (зима))
</div>
</div>
<script src="..\..\..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\..\..\sorter.js"></script>
</body>
</html>
