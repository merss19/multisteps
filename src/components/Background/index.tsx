import * as React from 'react'
import {bg} from './data'

class Background extends React.Component<{}, {}> {
	render() {
		return (
			<div className='entry__bg'>
				{bg}
			</div>
		)
	}
}

export default Background