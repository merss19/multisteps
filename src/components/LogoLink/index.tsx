import * as React from 'react'
import {browserHistory} from 'react-router'
import Cookie from '../../tools/cookie'

const LogoLink: React.SFC<{}> = () => (
	<svg className='svg-icon ys-logo-web'
		 onClick={() => {
			 Cookie().remove('token', {path: '/'})
			 Cookie().remove('txId', {path: '/'})
			 Cookie().remove('role', {path: '/'})
			 Cookie().remove('program', {path: '/'})
			 Cookie().remove('packageType', {path: '/'})
			 Cookie().remove('promoName', {path: '/'})
			 browserHistory.push('/')
		 }}>
		<use xlinkHref='#ys-logo-web'/>
	</svg>)

export default LogoLink
