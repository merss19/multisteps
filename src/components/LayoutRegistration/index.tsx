import LayoutRegistration from './main'
export {LayoutRegistration}