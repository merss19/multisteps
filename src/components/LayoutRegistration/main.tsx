import * as React from 'react'
import Background from '../Background'
import LogoLink from '../LogoLink'
import * as SVGInjector from 'svg-injector'
import {host} from '../../config'

interface Sprite {
	readonly body: any
	readonly img: HTMLImageElement
}

interface OwnProps {
	notLogo?: boolean
}
let url = process.env.NODE_ENV === 'test' ? host : '/'

class LayoutRegistration extends React.Component<OwnProps, {}> {

	componentDidMount() {
		this.svgInject()
	}

	svgInject(): void {
		const sprite: Sprite = {
			body: document.querySelector('body'),
			img : document.createElement('img')
		}
		sprite.img.className = 'injected-svg'
		sprite.img.id = 'svg-inject-me'
		sprite.img.src = url + 'src/assets/img/symbol-sprite.svg'
		sprite.body.appendChild(sprite.img)
		let mySVGsToInject: any = document.querySelector('#svg-inject-me')
		SVGInjector(mySVGsToInject)
	}

	renderLogo() {
		if (!this.props.notLogo) {
			return (
				<div className='grid entry-header'>
					<div className='grid__cell todayme-logo'>
						<LogoLink/>
					</div>
				</div>
			)
		}
		return null
	}

	render() {
		const {children} = this.props
		return (
			<div className='layout layout--entry'>
				{this.renderLogo()}
				{children}
				<Background />
			</div>
		)
	}
}

export default LayoutRegistration