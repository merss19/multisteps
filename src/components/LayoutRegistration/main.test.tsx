import {shallow, mount} from 'enzyme'
import LayoutRegistration from './main'
import * as React from 'react'
import toJson from 'enzyme-to-json'
import Background from '../Background'
import LogoLink from '../LogoLink'
import * as sinon from 'sinon'

process.env.NODE_ENV = 'test'

describe('Component LayoutRegistration', () => {
	const component = shallow(<LayoutRegistration/>)
	it('renders component', () => {
		expect(component).toHaveLength(1)
	})
	it('renders correctly', () => {
		const tree = toJson(shallow(<LayoutRegistration/>))
		expect(tree).toMatchSnapshot()
	})
	it('renders the LogoLink', () => {
		expect(component.find(LogoLink).exists()).toBe(true)
	})
	it('renders the Background', () => {
		expect(component.find(Background).exists()).toBe(true)
	})
})
describe('Component LayoutRegistration DOM', () => {
	const component = mount(<LayoutRegistration>
		<div>test</div>
	</LayoutRegistration>)
	it('children component defined', () => {
		expect(component.props().children).toBeDefined()
	})
	it('do not render logo', () => {
		const component = mount(<LayoutRegistration notLogo={true} />)
		//console.log(component.debug())
		expect(component.find(LogoLink).exists()).toBe(false)
	})
	it('calls componentDidMount', () => {
		const spy = sinon.spy(LayoutRegistration.prototype, 'componentDidMount')
		//const wrapper = mount(<LayoutRegistration />)
		expect(spy.calledOnce).toEqual(true)
	})
	it('calls svjInject', () => {
		const spy = sinon.spy(LayoutRegistration.prototype, 'svgInject')
		//const wrapper = mount(<LayoutRegistration />)
		expect(spy.calledOnce).toEqual(true)
	})
})