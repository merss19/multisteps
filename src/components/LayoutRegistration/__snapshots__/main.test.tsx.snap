// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LayoutRegistration renders correctly 1`] = `
<div
  className="layout layout--entry"
>
  <div
    className="grid entry-header"
  >
    <div
      className="grid__cell todayme-logo"
    >
      <LogoLink />
    </div>
  </div>
  <Background />
</div>
`;
